src/test/javascript/spec/component/exam/manage/exam-students.component.spec.ts

```diff
@@ -0,0 +1,146 @@
+const expect = chai.expect;
+
+describe('ExamStudentsComponent', () => {
+    const examCourse = { id: 1 } as Course;
+    const user1 = { id: 1 } as User;
+    const user2 = { id: 2, login: 'user2' } as User;
+
+    const route = ({
+        snapshot: { paramMap: convertToParamMap({ courseId: examCourse.id }) },
+        url: new Observable<UrlSegment[]>(),
+        data: { subscribe: (fn: (value: any) => void) => fn({ exam }) },
+    } as any) as ActivatedRoute;
+
+    let component: ExamStudentsComponent;
+    let fixture: ComponentFixture<ExamStudentsComponent>;
+    let examManagementService: ExamManagementService;
+    let userService: UserService;
+    let exam: Exam;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [ArtemisTestModule, NgxDatatableModule, RouterTestingModule],
+            declarations: [
+                ExamStudentsComponent,
+                MockComponent(StudentsExamImportButtonComponent),
+                MockComponent(DataTableComponent),
+                MockComponent(AlertComponent),
+                MockDirective(JhiTranslateDirective),
+                MockDirective(DeleteButtonDirective),
+            ],
+            providers: [{ provide: ActivatedRoute, useValue: route }],
+        }).compileComponents();
+
+        fixture = TestBed.createComponent(ExamStudentsComponent);
+        component = fixture.componentInstance;
+        examManagementService = TestBed.inject(ExamManagementService);
+        userService = TestBed.inject(UserService);
+
+        exam = { course: examCourse, id: 2, registeredUsers: [user1, user2] } as Exam;
+    });
+
+    afterEach(function () {
+        sinon.restore();
+    });
+
+    it('should initialize', () => {
+        fixture.detectChanges();
+        expect(component).to.be.ok;
+        expect(component.courseId).to.equal(examCourse.id);
+        expect(component.exam).to.deep.equal(exam);
+        expect(component.isLoading).to.be.false;
+    });
+
+    it('should handle auto-complete for user without login', () => {
+        const callbackSpy = sinon.spy();
+        fixture.detectChanges();
+
+        component.onAutocompleteSelect(user1, callbackSpy);
+        fixture.detectChanges();
+
+        expect(callbackSpy).to.have.been.calledOnceWith(user1);
+    });
+
+    it('should handle auto-complete for unregistered user', () => {
+        const user3 = { id: 3, login: 'user3' } as User;
+        const callbackSpy = sinon.spy();
+        const flashSpy = sinon.spy(component, 'flashRowClass');
+        const examServiceStub = sinon.stub(examManagementService, 'addStudentToExam').returns(of(new HttpResponse()));
+        fixture.detectChanges();
+
+        component.onAutocompleteSelect(user3, callbackSpy);
+        fixture.detectChanges();
+
+        expect(examServiceStub).to.have.been.calledOnceWith(examCourse.id, exam.id, user3.login);
+        expect(component.allRegisteredUsers).to.deep.equal([user1, user2, user3]);
```

Comment:
Not sure if deep equal impacts performance, but just fyi I think it would suffice to check the ids of the users instead of entire objects

Reply:
The objects are not large at all, so I doubt we'd see any noticeable performance gain by not using `deep` here. Functionality wise it shouldn't really matter, so I can remove it if you think without `deep` is better.

Reply:
I think you can leave it :)